src/game/server/instagib/round_stats.cpp

```diff
@@ -123,41 +123,61 @@ void IGameController::GetRoundEndStatsStrPsv(char *pBuf, size_t Size)
 	int ScoreLimit = m_GameInfo.m_ScoreLimit;
 	int TimeLimit = m_GameInfo.m_TimeLimit;
 
-	std::optional<std::string> RedClan = std::nullopt;
-	std::optional<std::string> BlueClan = std::nullopt;
+	char *RedClan = nullptr;
+	char *BlueClan = nullptr;
 
 	for(const CPlayer *pPlayer : GameServer()->m_apPlayers)
 	{
 		if(!pPlayer)
 			continue;
 
-		if(pPlayer->GetTeam() == TEAM_RED)
+		if(pPlayer->GetTeam() == TEAM_RED) // gof of c++, please forgive me for my sins
 		{
-			if(!RedClan.has_value())
+			if(strlen(Server()->ClientClan(pPlayer->GetCid())) == 0)
 			{
-				RedClan = Server()->ClientClan(pPlayer->GetCid());
+				free(RedClan);
+				RedClan = nullptr;
+				break;
+			}
+
+			if(!RedClan)
+			{
+				const char *NewClan = Server()->ClientClan(pPlayer->GetCid());
+				RedClan = (char *)calloc(strlen(NewClan) + 1, sizeof(char));
```

Comment:
You don't need to allocate or free any memory here. You can just point to the same memory.
The `Server()->ClientClan(pPlayer->GetCid())` will still be in scope and valid memory when its used further down in the `str_format` which then copies it.

You can just say
```
pRedClan = Server()->ClientClan(pPlayer->GetCid());
```